src/cdk-experimental/listbox/listbox.ts

```diff
@@ -194,360 +181,541 @@ export class CdkOption<T = unknown> implements ListKeyManagerOption, Highlightab
+   * @param event The focusout event
+   */
+  protected _handleFocusOut(event: FocusEvent) {
+    const otherElement = event.relatedTarget as Element;
+    if (this.element !== otherElement && !this.element.contains(otherElement)) {
+      this._onTouched();
     }
   }
 
-  /** Updates the key manager's active item to the given option. */
-  setActiveOption(option: CdkOption<T>) {
-    this._listKeyManager.updateActiveItem(option);
-    this._updateActiveOption();
+  /** Get the id of the active option if active descendant is being used. */
+  protected _getAriaActiveDescendant(): string | null | undefined {
+    return this._useActiveDescendant ? this.listKeyManager?.activeItem?.id : null;
   }
 
-  /**
-   * Saves a callback function to be invoked when the select's value
-   * changes from user input. Required to implement ControlValueAccessor.
-   */
-  registerOnChange(fn: (value: T) => void): void {
-    this._onChange = fn;
+  /** Get the tabindex for the listbox. */
+  protected _getTabIndex() {
+    if (this.disabled) {
+      return -1;
+    }
+    return this.useActiveDescendant || !this.listKeyManager.activeItem ? 0 : -1;
   }
 
-  /**
-   * Saves a callback function to be invoked when the select is blurred
-   * by the user. Required to implement ControlValueAccessor.
-   */
-  registerOnTouched(fn: () => {}): void {
-    this._onTouched = fn;
-  }
+  /** Initialize the key manager. */
+  private _initKeyManager() {
+    this.listKeyManager = new ActiveDescendantKeyManager(this.options)
+      .withWrap()
+      .withTypeAhead()
+      .withHomeAndEnd()
+      .withAllowedModifierKeys(['shiftKey']);
 
-  /** Sets the select's value. Required to implement ControlValueAccessor. */
-  writeValue(values: T | T[]): void {
-    if (this._options) {
-      this._setSelectionByValue(values);
+    if (this.orientation === 'vertical') {
+      this.listKeyManager.withVerticalOrientation();
+    } else {
+      this.listKeyManager.withHorizontalOrientation(this._dir?.value || 'ltr');
     }
+
+    this.listKeyManager.change
+      .pipe(takeUntil(this.destroyed))
+      .subscribe(() => this._focusActiveOption());
   }
 
-  /** Disables the select. Required to implement ControlValueAccessor. */
-  setDisabledState(isDisabled: boolean): void {
-    this.disabled = isDisabled;
+  // TODO(mmalerba): Should not depend on combobox.
+  private _updatePanelForSelection(option: CdkOption<T>) {
+    if (this._combobox) {
+      if (!this.multiple) {
+        this._combobox.updateAndClose(option.isSelected() ? option.value : []);
+      } else {
+        this._combobox.updateAndClose(this.value);
+      }
+    }
   }
 
-  /** Returns the values of the currently selected options. */
-  getSelectedValues(): T[] {
-    return this._options.filter(option => option.selected).map(option => option.value);
+  /** Update the selection mode when the 'multiple' property changes. */
+  private _updateSelectionModel() {
+    this.selectionModelSubject.next(
```

Comment:
I don't know if this is new or I missed it the first time around, but it seems weird to be swapping out selection models. If it's because `multiple` can't be changed, we can update the selection model instead.

Reply:
I'd rather leave it as is for this PR, we could look into that as a followup. `SelectionModel` is used by existing code so I want to minimize changes to it here to make it easier to land this PR